azure-quantum/tests/unit/test_optimization.py

```diff
@@ -574,6 +621,27 @@ def test_SimulatedAnnealing_experimental_input_params(self):
         self.assertEqual(True, good.supports_grouped_terms())
         self.assertEqual(True, good.supports_protobuf())
 
+    def test_SimulatedAnnealing_legacy_input_params(self):
+        ws = self.create_workspace()
+
+        good = SimulatedAnnealing(ws, name = "microsoft.simulatedannealing.cpu.legacy", timeout=1011, seed=4321)
+        self.assertIsNotNone(good)
+        self.assertEqual(
+            "microsoft.simulatedannealing-parameterfree.cpu.legacy", good.name
```

Comment:
Those are test for correct switching target to param free target if no method parameters were provided. User would use name = "microsoft.simulatedannealing.cpu.legacy" , but methos will switch to corresponding param free target if there is no method parameters, or it will use this target if there is parameter like beta_start.

Reply:
Oh I see, got it, thanks for clarifying!